pygac_fdr/tests/test_end2end.py

```diff
@@ -0,0 +1,470 @@
+        cls.test_data_dir = config['test_data_dir']
+        cls.cleanup = config['cleanup']
+        cls.resume = config['resume']
+        cls.fast = config['fast']
+        cls.rtol = float(config['rtol'])
+        cls.atol = float(config['atol'])
+        cls.plot = config['plot']
+
+        # Set class attributes
+        cls.cfg_file = os.path.join(os.path.dirname(__file__), '../../etc/pygac-fdr.yaml')
+        cls.tle_dir = os.path.join(cls.test_data_dir, 'tle')
+        cls.input_dir = os.path.join(cls.test_data_dir, 'input')
+        cls.output_dir = None  # to be set by subclasses
+        cls.output_dir_ref = os.path.join(cls.test_data_dir, 'output_ref')
+
+    @classmethod
+    def _unzip_gac_files(cls, filenames_gz, output_dir):
+        filenames = []
+        for filename_gz in filenames_gz:
+            basename_gz = os.path.basename(filename_gz)
+            basename = os.path.splitext(basename_gz)[0]
+            filename = os.path.join(output_dir, os.path.basename(basename))
+            LOG.info('Decompressing {}'.format(filename_gz))
+            with gzip.open(filename_gz, 'rb') as f_in:
+                with open(filename, 'wb') as f_out:
+                    shutil.copyfileobj(f_in, f_out)
+            filenames.append(filename)
+        return filenames
+
+    @classmethod
+    def _run(cls, gac_files_gz, dbfile=None):
+        """Read GAC files and write calibrated scenes to netCDF.
+
+        If database file is given, also collect & update metadata.
+        """
+        if cls.resume:
+            # Resume testing with existing output files
+            LOG.info('Resuming test with existing output files')
+            return sorted(
+                glob.glob(os.path.join(cls.output_dir, '*.nc'))
+            )
+
+        # Prepare output directory
+        if not os.path.isdir(cls.output_dir):
+            os.makedirs(cls.output_dir)
+        cls._cleanup_output_dir()
+
+        # Unzip GAC files
+        gac_files = cls._unzip_gac_files(gac_files_gz, cls.output_dir)
+
+        # Read GAC files and write netCDF files
+        run = ['pygac-fdr-run',
+               '--cfg', cls.cfg_file,
+               '--output-dir', cls.output_dir,
+               '--tle-dir', cls.tle_dir,
+               '--verbose', '--log-all'] + gac_files
+        call_subproc(run)
+        nc_files = sorted(glob.glob(cls.output_dir + '/*.nc'))
+
+        if dbfile:
+            # Collect & complement metadata
+            collect = ['pygac-fdr-mda-collect', '--dbfile', dbfile, '--if-exists', 'replace',
+                       '--verbose'] + nc_files
+            call_subproc(collect)
+
+            # Update metadata
+            update = ['pygac-fdr-mda-update', '--dbfile', dbfile]
+            call_subproc(update)
+
+        return nc_files
+
+    @classmethod
+    def _cleanup_output_dir(cls):
+        if cls.cleanup:
+            for item in os.listdir(cls.output_dir):
+                item = os.path.join(cls.output_dir, item)
```

Comment:
consider using `glob.glob` instead of `os.listdir`, or `pathlib.Path.glob` to access the path directly.

Reply:
I would beg to differ here, glob and listdir have two distinct functions. In this case, listdir ensures all items of the directory are visited. Using glob, you would have to check `*` and `.*` for example.

Reply:
@mraspaud, conceptually, I agree that globbing is a search activity, while listdir is an explicit command. My point here is more about the reassignment of the loop iterator variable.
Maybe `pathlib.Path.iterdir` might be a good solution.